src/engine/source/builder/policy.hpp

```diff
@@ -0,0 +1,267 @@
+     * @param graphName Name of the subgraph.
+     * @param type Type of the assets in the subgraph.
+     * @throws std::runtime_error if the asset cannot be built.
+     */
+    void buildGraph(const std::vector<std::shared_ptr<Asset>>& assets, const std::string& graphName);
+
+    /**
+     * @brief Inject Filters into specific subgraph.
+     *
+     * If a filter references an asset, it is added as child of the asset, and the asset's
+     * children are added as children of the filter.
+     * Otherwise nohting is done.
+     *
+     * @param graphName Name of the subgraph.
+     */
+    void addFilters(const std::string& graphName);
+
+    void saveGraph(const std::vector<std::shared_ptr<Asset>>& assets, const std::string& name)
+    {
+        // Add input asset
+        m_graphs.push_back(std::make_pair<std::string, Graph<std::string, std::shared_ptr<Asset>>>(
+            std::string {name},
+            {std::string(name + "Input"), std::make_shared<Asset>(name + "Input", getAssetType(name))}));
+
+        // Save assets
+        std::transform(assets.begin(),
+                       assets.end(),
+                       std::inserter(m_assets, m_assets.begin()),
+                       [&](auto& asset) { return std::make_pair(asset->m_name, asset); });
+
+        buildGraph(assets, name);
+
+        // Add filters
+        addFilters(name);
+
+        // Check integrity
+        auto graphPos = std::find_if(
+            m_graphs.begin(), m_graphs.end(), [&name](const auto& graph) { return std::get<0>(graph) == name; });
+        for (auto& [parent, children] : std::get<1>(*graphPos).m_edges)
+        {
+            if (!std::get<1>(*graphPos).hasNode(parent))
+            {
+                std::string childrenNames;
+                for (auto& child : children)
+                {
+                    childrenNames += child + " ";
+                }
+                throw std::runtime_error(fmt::format("Error building environment \"{}\". Asset \"{}\" requested for "
+                                                     "parent \"{}\" which could not be found",
+                                                     name,
+                                                     parent,
+                                                     childrenNames));
+            }
+            for (auto& child : children)
+            {
+                if (!std::get<1>(*graphPos).hasNode(child))
+                {
+                    throw std::runtime_error(fmt::format("Asset \"{}\" could not be found", child));
+                }
+            }
+        }
+    }
+
+public:
+    Policy() = default;
+
+    // TODO: Remove injected catalog dependencies ?
+    /**
+     * @brief Construct a new Policy object
+     *
+     * @param jsonDefinition Json definition of the environment.
+     * @param storeRead Store read interface.
+     * @param registry Registry interface.
+     * @throws std::runtime_error if the environment cannot be built.
```

Comment:
Documentation should also be updated to 'policy'.

Reply:
done